core/jazz_splunk-kinesis-log-streamer/components/utils.js

```diff
@@ -55,47 +119,185 @@ var getSubInfo = function (message, patternStr, index) {
-      data.asset_identifier = payload.logGroup.split(global_config.PATTERNS.asset_identifier_key)[1];
-      data.environment = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_environment, 2);
-      if (data.environment === "dev") {
-        let dev_environment = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_environment_dev, 2);
-        serviceInfo = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_environment_dev, 1);
-        data.environment = dev_environment;
+    var service_api_options = {
+      url: `${config.SERVICE_API_URL}${config.SERVICE_URL}?domain=${serviceParts[1]}&service=${serviceParts[2]}&environment=${serviceParts[serviceParts.length - 1]}`,
+      headers: {
+        "Content-Type": "application/json",
+        "Authorization": authToken
+      },
+      method: "GET",
+      rejectUnauthorized: false,
+      requestCert: true,
+      async: true
+    };
+
+    request(service_api_options, (error, response, body) => {
+      if (error) {
+        reject(error);
       } else {
-        serviceInfo = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_domain_service, 1);
+        if (response.statusCode && response.statusCode === 200) {
+          var responseBody = JSON.parse(body);
+          logger.debug("Response Body of Service Metadata is: " + JSON.stringify(responseBody));
+          resolve(responseBody.data.services[0])
+        } else {
+          logger.debug("Service not found for this service, domain, environment: ", JSON.stringify(service_api_options));
+          resolve([])
+        }
       }
+    })
+  });
+}
 
-      domainAndservice = serviceInfo;
-      if (serviceInfo.indexOf(global_config.PATTERNS.sls_app_function) !== -1) { // if yes then sls-app function
-        let serviceInfoArr = serviceInfo.split(global_config.PATTERNS.sls_app_function);
-        domainAndservice = serviceInfoArr[0];
-      }
+// Function to check if account is primary or not
+function checkIsPrimary(accountId, jsonConfig) {
+  var data = jsonConfig.config.AWS.ACCOUNTS;
+  var index = data.findIndex(x => x.ACCOUNTID == accountId);
+  if (data[index].PRIMARY) {
+    return data[index].PRIMARY;
+  } else {
+    return false;
+  }
+}
+
+// Function to get roleArn for specific accountId
+function getRolePlatformService(accountId, jsonConfig) {
+  var data = jsonConfig.config.AWS.ACCOUNTS;
+  var index = data.findIndex(x => x.ACCOUNTID == accountId);
+  return data[index].IAM.PLATFORMSERVICES_ROLEID;
+}
 
-      logger.debug("domainAndservice: " + domainAndservice)
+// Function to get configDB data
+function getConfigJson(config, token) {
+  return new Promise((resolve, reject) => {
+    var config_json_api_options = {
+      url: `${config.SERVICE_API_URL}${config.CONFIG_URL}`,
+      headers: {
+        "Content-Type": "application/json",
+        "Authorization": token
+      },
+      method: "GET",
+      rejectUnauthorized: false,
+      requestCert: true,
```

Comment:
Do we need these two flags - `requestCert` & `async`?

Reply:
No. Removed.